default/python/universe_generation/galaxy.py

```diff
@@ -379,27 +380,6 @@ def calc_universe_width(shape, size):
     return width
 
 
-def get_systems_within_jumps(origin_system, jumps):
-    """
-    Returns all systems within jumps jumps of system origin_system (including origin_system).
-    If jumps is 0, return list that only contains system origin_system.
-    If jumps is negative, return empty list.
-    """
-    if jumps < 0:
-        return []
-    matching_systems = [origin_system]
-    next_origin_systems = [origin_system]
-    while jumps > 0:
-        origin_systems = list(next_origin_systems)
-        next_origin_systems = []
-        for system in origin_systems:
-            neighbor_systems = [s for s in fo.sys_get_starlanes(system) if s not in matching_systems]
-            next_origin_systems.extend(neighbor_systems)
-            matching_systems.extend(neighbor_systems)
-        jumps -= 1
-    return matching_systems
```

Comment:
Moving that function out of the `galaxy` module has been proposed before, and while I can see the point, I wonder if it's really reasonable to create an extra module just for one function. The `galaxy` module has been named "galaxy" for this reason, because it originally was supposed to contain all kind of map related stuff, and not only the actual system placement functions.

Once we have more strictly topology related stuff, splitting that from `galaxy` makes more sense IMO.